### PR TITLE
[12.0] sale_commission: add analysis view for sale orders

### PR DESCRIPTION
**Pull request by eLBati**

### CODE REVIEW

**Comment by pedrobaeza** [comment]
What I miss here:

- Record rules for applying same permissions as in sale.order for "User: See own documents only".
- This is something not very valid, as final settlement depends on invoices, not on sales orders, so you can have a very old sales order, but not being invoiced until a lot, and the amounts will be distorted by this. My idea was to bring invoice data here to Sales section, but with the corresponding permissions.

**Comment by eLBati** [comment]
> Record rules for applying same permissions as in sale.order for "User: See own documents only".

Thanks, done

> This is something not very valid, as final settlement depends on invoices, not on sales orders, so you can have a very old sales order, but not being invoiced until a lot, and the amounts will be distorted by this

The purpose of the analysis based on sales is  more about forecast indeed; basically: how much will the company have to pay, based on (confirmed) orders?

**Comment by pedrobaeza** [comment]
Well, as said, that forecast can be very distorted due to the differences in dates from quotation/sales order to invoice.

**Comment by eLBati** [comment]
Even if distorted, it is anyway useful information that our customers ask for (consciously of the possible distortion)

**Comment by pedrobaeza** [comment]
OK, just add the information on the README with this warning (and thanks for the contribution). FYI, I have budgeted to a customer for the other information I'm talking about.

**Comment by eLBati** [comment]
@pedrobaeza added, thanks

**Comment by eLBati** [comment]
Another minor functionality added: 75ae859e4a6f41d939905a1cace6171477e3dccb

**Comment by pedrobaeza** [comment]
Please split each feature in a PR. For the second one, please use a computed searchable field (with `search` method) instead of storing it, as it will be slow to update the module if not, and it's a waste of records for a non usual thing.

**Comment by eLBati** [comment]
Moved the second commit to https://github.com/OCA/commission/pull/280

File: sale_commission/report/sale_commission_analysis_report.py [hunk 1]
```diff
@@ -107,3 +107,104 @@ def init(self):
                 AsIs(self._group_by())
             ),
         )
+
+
+class SaleOrderCommissionAnalysisReport(models.Model):
```
Comment (pedrobaeza) [inline sale_commission/report/sale_commission_analysis_report.py]: Better to put this on a new file, as they are separate models.

File: sale_commission/report/sale_commission_analysis_report_view.xml [hunk 2]
```diff
@@ -63,9 +63,74 @@
         <field name="view_mode">pivot,graph</field>
         <field name="search_view_id" ref="view_sale_commission_analysis_search"/>
         <field name="view_id" ref="view_sale_commission_analysis_graph"/>
-        <field name="help">This report performs analysis on your commissions added in invoice lines. You can check the amount and lines that will be settled by date, product, product category, aget, etc. Use this report to perform analysis on invoice lines agents not beeing settled yet.</field>
+        <field name="help">This report performs analysis on your commissions added in invoice lines. You can check the amount and lines that will be settled by date, product, product category, agent, etc. Use this report to perform analysis on invoice lines agents not beeing settled yet.</field>
     </record>
 
     <menuitem action="action_sale_commission_analysis_report" id="menu_sale_commissions_analysis" parent="account.account_reports_management_menu" sequence="15"/>
 
+    <record id="view_sale_order_commission_analysis_pivot" model="ir.ui.view">
```
Comment (pedrobaeza) [inline sale_commission/report/sale_commission_analysis_report_view.xml]: The same for this.

File: sale_commission/security/ir.model.access.csv [hunk 3]
```diff
@@ -14,3 +14,4 @@ access_sale_commission_settlement_line_manager,access_sale_commission_settlement
 access_sale_commission_settlement_line_salesman,access_sale_commission_settlement_line_salesman,model_sale_commission_settlement_line,sales_team.group_sale_salesman,1,0,0,0
 access_sale_commission_settlement_line_user,access_sale_commission_settlement_line_user,model_sale_commission_settlement_line,base.group_user,1,0,0,0
 access_sale_commission_analysis_report,access_sale_commission_analysis_report,model_sale_commission_analysis_report,base.group_user,1,0,0,0
+access_sale_order_commission_analysis_report,access_sale_order_commission_analysis_report,model_sale_order_commission_analysis_report,base.group_user,1,0,0,0
```
Comment (pedrobaeza) [inline sale_commission/security/ir.model.access.csv]: Shouldn't be sale user?

File: sale_commission/report/sale_commission_analysis_report.py (hunk 1 above)
Comment (eLBati) [reply sale_commission/report/sale_commission_analysis_report.py]: done

File: sale_commission/report/sale_commission_analysis_report_view.xml (hunk 2 above)
Comment (eLBati) [reply sale_commission/report/sale_commission_analysis_report_view.xml]: done

File: sale_commission/security/ir.model.access.csv (hunk 3 above)
Comment (eLBati) [reply sale_commission/security/ir.model.access.csv]: Modified to `sales_team.group_sale_salesman`

**Comment by mrcast** [comment]
@eLBati :
- create a customer
- create an agent
- associate agent on customer
- create a sale order with a line with e new product
- saving, I obtain the warning in attachment and I can't test the PR
- 
image

**Comment by mrcast** [comment]
@eLBati : Sorry, it was my mistake, don't consider my previous message

**Review by mrcast — APPROVED** [review]
Functional test, ok

**Comment by pedrobaeza** [comment]
/ocabot merge major

Now it needs to be ported to v13

**Pull request merged**